plugins/devices/adm3A-terminal/src/main/java/net/sf/emustudio/devices/adm3a/impl/LoadCursorPosition.java

```diff
@@ -19,135 +19,65 @@
-        ESC_CURSOR_CODES['T'] = 52;
-        ESC_CURSOR_CODES['U'] = 53;
-        ESC_CURSOR_CODES['V'] = 54;
-        ESC_CURSOR_CODES['W'] = 55;
-        ESC_CURSOR_CODES['X'] = 56;
-        ESC_CURSOR_CODES['Y'] = 57;
-        ESC_CURSOR_CODES['Z'] = 58;
-        ESC_CURSOR_CODES['['] = 59;
-        ESC_CURSOR_CODES['\\'] = 60;
-        ESC_CURSOR_CODES[']'] = 61;
-        ESC_CURSOR_CODES['^'] = 62;
-        ESC_CURSOR_CODES['_'] = 63;
-        ESC_CURSOR_CODES['`'] = 64;
-        ESC_CURSOR_CODES['a'] = 65;
-        ESC_CURSOR_CODES['b'] = 66;
-        ESC_CURSOR_CODES['c'] = 67;
-        ESC_CURSOR_CODES['d'] = 68;
-        ESC_CURSOR_CODES['e'] = 69;
-        ESC_CURSOR_CODES['f'] = 70;
-        ESC_CURSOR_CODES['g'] = 71;
-        ESC_CURSOR_CODES['h'] = 72;
-        ESC_CURSOR_CODES['i'] = 73;
-        ESC_CURSOR_CODES['j'] = 74;
-        ESC_CURSOR_CODES['k'] = 75;
-        ESC_CURSOR_CODES['l'] = 76;
-        ESC_CURSOR_CODES['m'] = 77;
-        ESC_CURSOR_CODES['n'] = 78;
-        ESC_CURSOR_CODES['o'] = 79;
-    }
-
+    // Escape sequence byte
+    private static final int ASCII_ESC = 0x1B;
+    // Offset in ASCII for X and Y coordinate
+    private static final int ASCII_COORDINATE_OFFSET = 32;
     enum ExpectedSequence {
-        ESCAPE, ASSIGN, XY
+        ESCAPE, ASSIGN, X, Y
     }
 
     private final Cursor cursor;
 
     private volatile ExpectedSequence expect = ESCAPE;
+    private int cursorY;
 
     LoadCursorPosition(Cursor cursor) {
         this.cursor = Objects.requireNonNull(cursor);
     }
 
-    boolean accept(Short data) {
-        short ESC = 0x1B;
-        boolean accepted = false;
+    /**
+     * Checks bounds of the X and Y variables
+     *
+     * @param data received char
+     * @return true if in bounds
+     */
+    private boolean checkBounds(Short data) {
+        return data >= ' ' && data <= 'o';
+    }
 
-        if (expect == ESCAPE && data == ESC) {
+    boolean accept(Short data) {
+        if (expect == ESCAPE && data == ASCII_ESC) {
             expect = ASSIGN;
-            accepted = true;
+            return true;
         } else if (expect == ASSIGN && data == '=') {
-            expect = XY;
-            accepted = true;
-        } else if (expect == XY) {
-            int cursorPos = ESC_CURSOR_CODES[data];
-            if (cursorPos != -1) {
-                if (cursorPos < 24) {
-                    cursor.set(cursorPos, cursorPos);
-                } else {
-                    cursor.set(cursorPos, cursor.getPoint().y);
-                }
+            expect = Y;
+            return true;
+        } else if (expect == Y) {
+            if (!checkBounds(data)) {
+                expect = ESCAPE;
+                return true;
             }
+
+            cursorY = data - ASCII_COORDINATE_OFFSET;
+            expect = X;
+            return true;
+        } else if (expect == X) {
+            int cursorX = data - ASCII_COORDINATE_OFFSET;
+
+            if (checkBounds(data)) {
```

Comment:
Here maybe the same thing - if `checkBounds` returns `false`, then the sequence is not accepted and the method should return `false`.